nodejs/eks/core.ts

```diff
@@ -0,0 +1,372 @@
+ */
+export interface RoleMapping {
+    /**
+     * The ARN of the IAM role to add.
+     */
+    roleArn: pulumi.Input<aws.ARN>;
+
+     /**
+     * The user name within Kubernetes to map to the IAM role. By default, the user name is the ARN of the IAM role.
+     */
+    username: pulumi.Input<string>;
+
+     /**
+     * A list of groups within Kubernetes to which the role is mapped.
+     */
+    groups: pulumi.Input<pulumi.Input<string>[]>;
+}
+
+ /**
+ * UserMapping describes a mapping from an AWS IAM user to a Kubernetes user and groups.
+ */
+export interface UserMapping {
+    /**
+     * The ARN of the IAM user to add.
+     */
+    userArn: pulumi.Input<aws.ARN>;
+
+     /**
+     * The user name within Kubernetes to map to the IAM user. By default, the user name is the ARN of the IAM user.
+     */
+    username: pulumi.Input<string>;
+
+     /**
+     * A list of groups within Kubernetes to which the user is mapped to.
+     */
+    groups: pulumi.Input<pulumi.Input<string>[]>;
+}
+
+/**
+ * CoreOptions describes the configuration options accepted by a Core component.
+ */
+export interface CoreOptions {
+    /**
+     * The VPC in which to create the cluster. If unset, the cluster will be created in the default VPC.
+     */
+    vpcId?: pulumi.Input<string>;
+
+    /**
+     * The subnets to attach to the EKS cluster. If either vpcId or subnetIds is unset, the cluster will use the
+     * default VPC's subnets.
+     */
+    subnetIds?: pulumi.Input<pulumi.Input<string>[]>;
+
+    /**
+     * Optional mappings from AWS IAM roles to Kubernetes users and groups.
+     */
+    roleMappings?: pulumi.Input<pulumi.Input<RoleMapping>[]>;
+
+     /**
+     * Optional mappings from AWS IAM users to Kubernetes users and groups.
+     */
+    userMappings?: pulumi.Input<pulumi.Input<UserMapping>[]>;
+
+    /**
+    * The instance role to use for all nodes in this workder pool.
+    */
+    instanceRole?: pulumi.Input<aws.iam.Role>;
+
+    /**
+     * An optional set of StorageClasses to enable for the cluster. If this is a single volume type rather than a map,
+     * a single StorageClass will be created for that volume type and made the cluster's default StorageClass.
+     */
+    storageClasses?: { [name: string]: StorageClass } | EBSVolumeType;
+
+    /**
+     * The configiuration of the Amazon VPC CNI plugin for this instance. Defaults are described in the documentation
+     * for the VpcCniOptions type.
+     */
+    vpcCniOptions?: VpcCniOptions;
+}
+
+/**
+ * Core is a component that wraps the core AWS and Kubernetes resources necessary to run an EKS cluster and its optional
+ * StorageClasses.
+ *
+ * Note that the EKS cluster created by this component will not have any worker nodes, so deploying many important
+ * Kubernetes resources will fail. Worker nodes can be created with the WorkerPool component.
+ */
+export class Core extends pulumi.ComponentResource {
```

Comment:
I'm not completely sure about the name `Core` here.  And more generally, it seems like we may not need a separate component here.  It might be nicer to make `eks.Cluster` not require any of the other stuff (or allow explicitly asking to not include worker nodes, etc.) so that the `eks.Cluster` component and the `eks.Core` component become the same thing (just with different arguments).

That point is related to making sure we have a really nice interface to `eks.Cluster` to take advantage of using existing worker groups.

Reply:
For this PR in particular though, feels like we may want to not expose this component, and wait for a follow up focused on improving the `eks.Cluster` API and see if we really need an additional component here when we do that API design?

Reply:
> It might be nicer to make eks.Cluster not require any of the other stuff (or allow explicitly asking to not include worker nodes, etc.)

My original PR (#22) includes this change and I have rebased that PR on top of this branch: https://github.com/houqp/pulumi-eks/commit/c0c204c7de65a5e9673ab2267ca7146326a1091e. The full rebase is at: https://github.com/houqp/pulumi-eks/commits/improvement. Once this refactoring is merged, I am going to send a PR for that branch.

So in short, setting `deployDashboard` to `true` and `skipDefaultWorkerPool` to `true` will effectively create an `eks.Cluster` that's equivalent to `Core`.

@pgavlin what's your thought on this?